config-service-impl/src/main/java/org/hypertrace/config/service/store/DocumentConfigStore.java

```diff
@@ -254,6 +268,22 @@ private Map<ConfigResourceContext, Optional<ConfigDocument>> getLatestVersionCon
     return latestVersionConfigDocs;
   }
 
+  private Filter buildConfigResourceContextsFilter(
+      Set<ConfigResourceContext> configResourceContexts) {
+    List<Filter> childFilters =
+        configResourceContexts.stream()
+            .map(this::getConfigResourceFieldContextFilter)
+            .collect(Collectors.toUnmodifiableList());
+    Filter configResourceFieldContextFilter = new Filter();
+    configResourceFieldContextFilter.setOp(Filter.Op.OR);
```

Comment:
Note this will be incorrect for an empty set - it would build an "all" filter rather than a "none". You're checking outside, which is good, but you should check in here too so future usages can't make that mistake.

Reply:
If you're trying to build a filter that matches N number of IDs, and N is 0, it should return no results. Here it returns _all_ results in the 0 case. If we had implemented delete without an empty check, and someone called it with an empty list it would drop the whole collection (for that tenant/resource).

```
collection.delete(buildConfigResourceContextsFilter(configResourceContexts));
```

In other words, there's no exposed bug here, but that relies on callers doing validation which can easily be done in here.